pkg/apiserver/handlers_test.go

```diff
@@ -142,6 +145,57 @@ func TestReadOnly(t *testing.T) {
 	}
 }
 
+type fakeAuthenticator struct {
+	authorized bool
+	err        error
+}
+
+func (f *fakeAuthenticator) AuthenticateRequest(req *http.Request) (user.Info, bool, error) {
+	return nil, f.authorized, f.err
+}
+
+func TestRequestAuthentication(t *testing.T) {
```

Comment:
I think there's an integration test for this, since for authentication it's important to catch setup/construction errors as well as unit test type errors. @erictune what do you think about this test?